numpy/lib/recfunctions.py

```diff
@@ -948,6 +966,9 @@ def structured_to_unstructured(arr, dtype=None, copy=False, casting='unsafe'):
 
     fields = _get_fields_and_offsets(arr.dtype)
     n_fields = len(fields)
+    if n_fields == 0:
+        raise ValueError("arr must have at least one field: "
+                         "output dtype or shape are ambiguous")
```

Comment:
Can you persuade me that the output shape is ambiguous?

I'm thinking this should be:
```
if n_fields == 0 and dtypes is None:
   raise ValueError("Unable to guess dtype") # or your longer message
elif n_fields == 0:
   raise NotImplementedError  # too many bugs elsewhere for this to be possible
```